### ISSUE TITLE
Rest: message decoding

### ISSUE THREAD

**Issue by ricardopereira** [ably/ably-ios#97]
I have a **dictionary** and I publish it like:
```
dictionary: NSDictionary = @{ name = "John"; number = 3; }

channel.publish(dictionary)
```

and the lib sends it like (dictionary -> binary -> base64):
```
{
  "data": "eyJudW1iZXIiOjMsIm5hbWUiOiJKb2huIn0=",
  "encoding": "json/base64"
}
               ^~~~~~~~~~ bug?
```

I was expecting a different payload, something like (dictionary -> json):
```
{
  "data": {
    "name": "John",
    "number": 3
  },
  "encoding": "json"
}
```

The same happens with `NSArray`.

**References**

> (RSL4d3) a JSON Message payload is stringified either as a JSON Object or Array and represented as a JSON string and the encoding attribute is set to “json”

> (RSL4d4) All messages received will be decoded based on the encoding field and deliver payloads in the format they were sent in i.e. binary, string, or a structured type containing the parsed JSON

**Comment by mattheworiordan** [comment]
Yes, agreed, this should be JSON encoded as opposed to binary encoded.

**Comment by ricardopereira** [comment]
@mattheworiordan Fixed and test done 16448e5

**Comment by mattheworiordan** [comment]
Great, is that part of a PR?

**Comment by ricardopereira** [comment]
Yes #96

**Comment by ricardopereira** [comment]
I opened a PR: fix #122
